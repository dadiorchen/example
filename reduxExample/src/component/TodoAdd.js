//@flow
/* This component is to show a input and add todo to todo-list */
import React from "react";
import {connect} from 'react-redux'
import {bindActionCreators} from 'redux'
import todoModel from '../model/TodoModel.js'


type Props = {
	todoAddByContent : (content : string) => void,
}
export class TodoAdd extends React.Component<Props,{
	//state
	content	: string,
}>{
	constructor( props : Props){
		super(props);
		this.state = {
			content	: '',
		}
	}
	/********************** properties ************/
	/********************** react method ***********/
	/********************** component method *******/
	handleChange = (e : any) => {
		this.setState({
			content	: e.target.value,
		})
	}

	/* Add to model */
	handleAdd = () => {
		this.props.todoAddByContent(this.state.content)
		//clear the input
		this.setState({
			content	: ''
		})
	}

	render(){
		console.warn('+++render:todoadd')
		return(
			<div>
				<input 
					type='text' 
					placeholder='Please input todo content'
					value={this.state.content}
					onChange={this.handleChange}
				/>
				<button
					onClick={this.handleAdd}
				>
				Add
				</button>
			</div>
		)
	}
}

const CTodoAdd = connect(
	null,
//	(dispatch) => {
//		/* The original way */
////		return {
////			todoAddByContent : (content : string) => {
////				dispatch(todoModel.actions.todoAddByContent(content))
////			}
////		}
//		/* Using bindActionCreator */
////		return bindActionCreators({todoAddByContent:todoModel.actions.todoAddByContent },dispatch)
//	}
	/* The great way , Its VERY sample !! */
	{
			todoAddByContent : todoModel.actions.todoAddByContent
	}
)(TodoAdd)
export {CTodoAdd}

