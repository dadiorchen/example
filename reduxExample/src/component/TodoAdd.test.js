//@flow
import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import {TodoAdd} from './TodoAdd.js'

describe('TestTodoAdd',() => {
	it('TestTodoAdd',() => {
		const mock = jest.fn()
		const w = shallow(<TodoAdd 
				todoAddByContent={mock}
			/>)
		expect(toJson(w)).toMatchSnapshot()
		//click -> text is null
		w.setState({content:'text'})
		w.find('button').simulate('click')
		expect(w.state('content')).toBe('')
		expect(mock).toHaveBeenCalled()
	})
})
