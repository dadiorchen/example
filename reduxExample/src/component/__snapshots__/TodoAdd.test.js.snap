// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestTodoAdd TestTodoAdd 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Please input todo content"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
